flaml/automl.py

```diff
@@ -1085,14 +1128,25 @@ def custom_metric(
                 hyperparamter configurations for the corresponding estimators.
             seed: int or None, default=None | The random seed for np.random.
             **fit_kwargs: Other key word arguments to pass to fit() function of
-                the searched learners, such as sample_weight.
+                the searched learners, such as sample_weight. Include period and
+                freq as key word arguments for forecast task.
         '''
         self._start_time_flag = time.time()
         self._state.task = task
         self._state.log_training_metric = log_training_metric
         self._state.fit_kwargs = fit_kwargs
         self._state.weight_val = sample_weight_val
         self._state.groups = groups
+
+        if self._state.task == 'forecast':
+            if dataframe is not None:

```

Comment:
What happens if the dataframe is None? It is possible the user provided the X_train and y_train right?

Reply:
Yes, it is possible that the user provided the X_train and y_train, where X_train is the timestamps and y_train is the value needed to be predicted